test/test_foreach.py

```diff
@@ -127,27 +173,53 @@ def _test_bin_op_list_alpha(self, device, dtype, foreach_op, foreach_op_, torch_
             else:
                 self.assertEqual(tensors1, expected)
 
+    # note(mkozuki): fastpath test uses dtypes which fastpath implementation supports.
+    # To confirm the dtypes of `OpInfo` cover the dtypes that the function support,
+    # this test does not use `try-except` for fastpath.
+    def _regular_unary_test(self, dtype, op, ref, inputs, is_fastpath):
+        if is_fastpath:
+            self.assertEqual(ref(inputs), op(inputs, self.is_cuda, is_fastpath))
+            return
+        try:
+            actual = op(inputs, self.is_cuda, is_fastpath)
+        except RuntimeError as e:
+            with self.assertRaisesRegex(type(e), re.escape(str(e))):
+                ref(inputs)
+        else:
+            expected = ref(inputs)
+            self.assertEqual(actual, expected)
+
+    def _inplace_unary_test(self, dtype, inplace, inplace_ref, inputs, is_fastpath):
+        copied_inputs = [[t.clone().detach() for t in tensors] for tensors in inputs]
+        try:
+            inplace(inputs, self.is_cuda, is_fastpath)
```

Comment:
out of curiosity, what op throws an error here? In regular (not inplace) unary test you don't need this try-except block. Can you please leave a comment about it in the test?

Reply:
Slowpath test uses all dtypes including ones that regular functions do not support: for example, `log` and `log2` do not support fp16 CPU tensors.
Thus, I use `try-except` block to check that the behavior of foreach slowpath is the same as that of the counterpart regular function.

Reply:
Failing tests are acos,asin,atan,cos,cosh,exp,log10,log2,log,sin,sih,tan,tanh with CPU & integer types because of "result type Float can't be cast to the desired output type".

I guess `getOpInfo` function will make this simpler

Reply:
Ok, so it's due to inplace integer inputs, cool, thanks for the comment.